include/seqan3/io/detail/magic_header.hpp

```diff
@@ -85,6 +87,29 @@ struct bgzf_compression
     //  B       C       [SLEN        ]  [BSIZE       ]
         '\x42', '\x43', '\x02', '\x00', '\x00', '\x00'
     };
+
+    /*!\brief Checks if the given header is a bgzf header.
+     * \param[in] header The header to validate.
+     * \returns `true` if it is a bgzf header, otherwise `false`.
+     */
+    static bool validate_header(char const * header)
```

Comment:
Can we use `std::string_view` here? I think we should adopt that as policy everywhere instead of `char const *`...

Reply:
Pheww, I am not sure if this is actually what you want in this context. I would rather pass it as the original std::array data.